src/app/vm/vm-creation/vm-creation.component.ts

```diff
@@ -90,14 +90,18 @@ export class VmCreationComponent {
   }
 
   public get showResizeSlider(): boolean {
-    return this.vmCreationState.template
-      && !isTemplate(this.vmCreationState.template)
-      && this.showRootDiskResize
-      && !!this.vmCreationState.rootDiskMinSize;
+    return !!this.vmCreationState.template;
   }
 
   public get rootDiskSizeLimit(): number {
-    return this.account && this.account.primarystorageavailable;
+    const storageAvailable = this.account && this.account.primarystorageavailable;
+    if (storageAvailable.toString() === 'Unlimited' && this.maxRootDiskSize) {
+      return this.maxRootDiskSize
+    } else if (storageAvailable < this.maxRootDiskSize) {
+      return storageAvailable;
+    } else {
+      return this.maxRootDiskSize || storageAvailable;
```

Comment:
You do not need else if you use return.
If maxRootDiskSize is exist then why you use a `this.maxRootDiskSize || storageAvailable;`